src/docdb/tree/DocDBStoredProceduresTreeItem.ts

```diff
@@ -69,14 +71,29 @@ export class DocDBStoredProceduresTreeItem extends DocDBTreeItemBase<StoredProce
         return this.parent.getContainerClient(client);
     }
 
-    private validateName(name: string): string | null | undefined {
+    private async validateStoredProcedureName(name: string, getChildrenTask: Promise<AzExtTreeItem[]>): Promise<string | null | undefined> {
+        const currStoredProcedureList = await getChildrenTask;
+        const currStoredProcedureNames: string[] = [];
+        for (const sp of currStoredProcedureList) {
+            if (sp instanceof DocDBStoredProcedureTreeItem) {
+                currStoredProcedureNames.push(sp.id);
+            }
+        }
```

Comment:
I think it would be more efficient to do this work in `createChildImpl`. That way we only have to get the existing stored procedures once instead of every time `validateStoredProcedureName` is called